Wikipedia/Code/ArticleViewController.swift

```diff
@@ -114,7 +114,6 @@ class ArticleViewController: ViewController, HintPresenting {
     }
     
     deinit {
-        NotificationCenter.default.removeObserver(self)
```

Comment:
I could be wrong but I believe that strange build warning is related to some future Swift 6 `actor` isolation-related [work](https://github.com/apple/swift-evolution/blob/main/proposals/0371-isolated-synchronous-deinit.md).

Would you be comfortable adding `NotificationCenter.default.removeObserver(self)` back to the end of this function (which prevents the build warnings from displaying) or would you prefer to keep it as-is? My reasoning for suggesting we keep it is mostly preventative – if someone in the future _does_ for whatever reason decide to use the block-based API, we're opening ourselves up to an easy to miss potential memory leak in our most used view. In my mind, no harm is being overly protective here.

Reply:
Good idea! I didn't think about just swapping the order around. This is done.

It's strange to me that the call in the beginning would cause those other warnings to pop up. We're referencing properties in other view controller `deinit`s without warning, which is what clued me to something weird going on specifically in this one. Maybe your linked documentation will clear things up for me.